server/sockets/list.js

```diff
@@ -348,20 +384,245 @@ const emitRemoveMemberOnLeaveCohort = socket =>
+  cohortClients
+) =>
+  socket.on(ListActionTypes.REMOVE_MEMBER_SUCCESS, data => {
+    const { listId, userId } = data;
+
+    // Broadcast to that removed user if he is at dashboard view
+    if (dashboardClients.has(userId)) {
+      const { socketId } = dashboardClients.get(userId);
+
+      socket.broadcast
+        .to(socketId)
+        .emit(ListActionTypes.DELETE_SUCCESS, listId);
+    }
+
+    if (listClients.has(userId)) {
+      List.findById(listId)
+        .lean()
+        .exec()
+        .then(doc => {
+          if (doc) {
+            const { socketId, viewId } = listClients.get(userId);
+            const { cohortId } = doc;
+            const data = { listId };
+
+            if (viewId === listId) {
+              if (cohortId) {
+                data.cohortId = cohortId;
+              }
+
+              // Broadcast to that removed user if he is on the list view
+              socket.broadcast
+                .to(socketId)
+                .emit(ListActionTypes.REMOVE_BY_SOMEONE, data);
+            }
+          }
+        });
+    }
+
+    // Broadcast to that removed user if he is on the cohort view
+    if (cohortClients.has(userId)) {
+      const { socketId } = cohortClients.get(userId);
+
+      socket.broadcast
+        .to(socketId)
+        .emit(ListActionTypes.DELETE_SUCCESS, listId);
+    }
+
+    // Broadcast to every user on the list
+    socket.broadcast
+      .to(`sack-${listId}`)
+      .emit(ListActionTypes.REMOVE_MEMBER_SUCCESS, { listId, userId });
+  });
+
+const archiveList = (socket, dashboardClients, cohortClients, listClients) =>
+  socket.on(ListActionTypes.ARCHIVE_SUCCESS, data => {
+    const { listId, cohortId } = data;
+
+    List.findById(listId)
+      .populate('cohortId')
+      .lean()
+      .exec()
+      .then(doc => {
+        const { viewersIds, cohortId: cohort } = doc;
+        const list = { ...doc, cohortId };
+
+        viewersIds.forEach(viewerId => {
+          const id = viewerId.toString();
+          let isGuest = false;
+
+          if (cohort) {
+            const { memberIds: cohortMemberIds } = cohort;
+            isGuest = !checkIfArrayContainsUserId(cohortMemberIds, viewerId);
+          }
+
+          const dataToSend = { isGuest, ...data };
+
+          if (listClients.has(id)) {
+            const { socketId, viewId } = listClients.get(id);
+
+            if (viewId === listId) {
+              socket.broadcast
+                .to(socketId)
+                .emit(ListActionTypes.ARCHIVE_SUCCESS, dataToSend);
+            }
+          }
+
+          if (dashboardClients.has(id)) {
+            const { socketId } = dashboardClients.get(id);
+
+            // Broadcast to clients on dashboard that have this list
+            socket.broadcast
+              .to(socketId)
+              .emit(ListActionTypes.DELETE_SUCCESS, listId);
+
+            socket.broadcast
+              .to(socketId)
+              .emit(ListActionTypes.FETCH_ARCHIVED_META_DATA_SUCCESS, {
```

Comment:
check if emitting those to events is necessary if you emit only the second one you update the old list meta-data with the new meta-data so the list should disappear from because of the selector gets only not archived list...